src/rgw/rgw_acl_swift.cc

```diff
@@ -82,14 +82,14 @@ void RGWAccessControlPolicy_SWIFT::add_grants(RGWRados *store, list<string>& uid
   }
 }
 
-bool RGWAccessControlPolicy_SWIFT::create(RGWRados *store, rgw_user& id, string& name, string& read_list, string& write_list, uint32_t& rw_mask)
+bool RGWAccessControlPolicy_SWIFT::create(RGWRados *store, rgw_user& id, string& name, const char* read_list, const char* write_list, uint32_t& rw_mask)
```

Comment:
why not use string?

Reply:
Because std::string has no concept of a "NULL" or unset value.  That's the essence of the bug this commit fixes.